ckan/views/dataset.py

```diff
@@ -0,0 +1,1175 @@
+        u'user': g.user,
+        u'for_view': True,
+        u'auth_user_obj': g.userobj
+    }
+    data_dict = {u'id': id, u'include_tracking': True}
+
+    # check if package exists
+    try:
+        pkg_dict = get_action(u'package_show')(context, data_dict)
+        pkg = context[u'package']
+    except (NotFound, NotAuthorized):
+        return base.abort(404, _(u'Dataset not found'))
+
+    # FEATURE: [start] https://github.com/ckan/ckan/pull/3972
+    is_activity_archive = True
+    activity_id = request.args.get(u'activity_id')
+    if activity_id:
+        # view an 'old' version of the package, as recorded in the
+        # activity stream
+        try:
+            activity = get_action(u'activity_show')(
+                context, {
+                    u'id': activity_id
+                }
+            )
+        except NotFound:
+            return base.abort(404, _(u'Activity not found'))
+        except NotAuthorized:
+            return base.abort(403, _(u'Unauthorized to create a package'))
+        try:
+            pkg_dict = activity[u'data'][u'package']
+        except KeyError:
+            return base.abort(404, _(u'Dataset not found'))
+        # Earlier versions of CKAN only stored the package table in the
+        # activity, so add a placeholder for resources, or the template
+        # will crash.
+        pkg_dict.setdefault(u'resources', [])
+        is_activity_archive = True
+    # FEATURE: [end]
+
+    # can the resources be previewed?
+    for resource in pkg_dict[u'resources']:
+        resource_views = get_action(u'resource_view_list')(
+            context, {
+                u'id': resource[u'id']
+            }
+        )
+        resource[u'has_views'] = len(resource_views) > 0
+
+    package_type = pkg_dict[u'type'] or package_type
+    _setup_template_variables(context, {u'id': id}, package_type=package_type)
+
+    template = _get_pkg_template(u'read_template', package_type)
+    try:
+        return base.render(
+            template, {
+                u'dataset_type': package_type,
+                u'pkg_dict': pkg_dict,
+                u'pkg': pkg,
+                u'is_activity_archive': is_activity_archive
+            }
+        )
+    except TemplateNotFound as e:
+        msg = _(
+            u"Viewing datasets of type \"{package_type}\" is "
+            u"not supported ({file_!r}).".format(
+                package_type=package_type, file_=e.message
+            )
+        )
+        return base.abort(404, msg)
+
+    assert False, u"We should never get here"
+
+
+class CreateView(MethodView):
+    def _is_save(self):
+        return u'save' in request.form
+
+    def _prepare(self, data=None):
+
+        context = {
+            u'model': model,
+            u'session': model.Session,
+            u'user': g.user,
+            u'auth_user_obj': g.userobj,
+            u'save': self._is_save()
+        }
+
+        # Package needs to have a organization group in the call to
```

Comment:
I'm not following this comment and how it relates to call to `check_access` below